pkg/controller/apiextensions/crds/crd.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crds
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	kyaml "k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/crossplane/crossplane/apis/apiextensions/v1alpha1"
+)
+
+// CRDOption is used to manipulate base crd.
+type CRDOption func(*v1beta1.CustomResourceDefinition)
+
+// NOTE(muvaf): We use v1beta1.CustomResourceDefinition for backward compatibility
+// with clusters pre-1.16
+
+// BaseCRD returns a base template for generating a CRD.
+func BaseCRD(opts ...CRDOption) *v1beta1.CustomResourceDefinition {
+	falseVal := false
+	// TODO(muvaf): Add proper descriptions.
+	crd := &v1beta1.CustomResourceDefinition{
+		Spec: v1beta1.CustomResourceDefinitionSpec{
+			PreserveUnknownFields: &falseVal,
+			Subresources: &v1beta1.CustomResourceSubresources{
+				Status: &v1beta1.CustomResourceSubresourceStatus{},
+			},
+			Validation: &v1beta1.CustomResourceValidation{
+				OpenAPIV3Schema: &v1beta1.JSONSchemaProps{
+					Type: "object",
+					Properties: map[string]v1beta1.JSONSchemaProps{
+						"apiVersion": {
+							Type: "string",
+						},
+						"kind": {
+							Type: "string",
+						},
+						"metadata": {
+							// NOTE(muvaf): api-server takes care of validating
+							// metadata.
```

Comment:
Most CRDs still include a schema for metadata though, right? Could be worth including in future.